src/explore-education-statistics-frontend/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -24,156 +24,167 @@ interface Props {
   locations: LocationFilter[];
   results: TableData['result'];
   footnotes?: TableData['footnotes'];
+  tableHeadersConfig?: TableHeadersFormValues;
```

Comment:
Why is this optional? From what I can see, default values are being set in `TableToolPage`, so this shouldn't need to be. This way we can avoid the horrible conditional logic that we have to use everywhere currently.

Reply:
I wanted to keep the header config optional, so that data could be rendered with a default configuration if a configuration wasn't provided, keeping the table rendering component somewhat flexible.

We spoke about potentially storing configured charts against permalinks, keeping the tableHeaderConfig optional would allow to easily change the rendering of a stored chart permalink from a chart to a table.

Perhaps we should discuss further

Reply:
Fair enough, then provide a default value for `tableHeaders` instead then (this would be the same as `TableToolPage`)